### PR TITLE
Fix "generator" language in word2vec docs

### LINKED ISSUE #2934: Word2Vec docs indicate sents can be a generator, but it can't
#### Problem description

Tried to pass a generator as the `sentences` argument when training Word2Vec, as suggested [in the docs](https://radimrehurek.com/gensim/models/word2vec.html#usage-examples:~:text=sentences%20can%20be%20a%20generator). But then I get this error:

    TypeError: You can't pass a generator as the sentences argument. Try a sequence.

#### Steps/code/corpus to reproduce

```

from gensim.models import Word2Vec

def get_sents():
    for ii in range(100):
        yield ['hi']

sents = get_sents()
model = Word2Vec(sents, size=100, window=5, min_count=1, workers=4)
```

#### Versions

Please provide the output of:

```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import struct; print("Bits", 8 * struct.calcsize("P"))
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import gensim; print("gensim", gensim.__version__)
from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
```
```
Linux-5.8.5-arch1-1-x86_64-with-glibc2.2.5
Python 3.8.5 (default, Jul 27 2020, 08:42:51) 
[GCC 10.1.0]
Bits 64
NumPy 1.19.1
SciPy 1.4.1
gensim 3.8.3
FAST_VERSION 1
```

### PR DESCRIPTION
**Pull request by polm**
Docs say you can use a generator as the first argument, but you can't.

The tempfile path was also unused, so that's been removed.

Fixes #2934.

### CODE REVIEW

File: gensim/models/word2vec.py [hunk 1]
```diff
@@ -39,18 +39,15 @@
...
     >>> from gensim.models import Word2Vec
     >>>
-    >>> path = get_tmpfile("word2vec.model")
-    >>>
     >>> model = Word2Vec(common_texts, size=100, window=5, min_count=1, workers=4)
     >>> model.save("word2vec.model")
 
-The training is streamed, meaning `sentences` can be a generator, reading input data
-from disk on-the-fly, without loading the entire corpus into RAM.
+`common_texts` is a list of lists of strings, where each list is a document.
```
Comment (piskvorky) [inline gensim/models/word2vec.py:48]: This is not good – the streaming is essential. We've had many instances where people read the docs and assume the input **must** be a list, which is not the case.

Please keep the streaming (iterable) language in, just drop the generator.

**Review by piskvorky — CHANGES_REQUESTED** [review]

File: gensim/models/word2vec.py (hunk 1 above)
Comment (polm) [reply gensim/models/word2vec.py:48]: Ah OK, I'll fix that.

**Comment by polm** [comment]
For reference, it looks like people have been confused about this for years :/ 

https://stackoverflow.com/questions/34166369/generator-is-not-an-iterator

**Comment by piskvorky** [comment]
Yes. There was a point in history where a generator was really accepted (and it still is in other Gensim models), which doesn't help.

Thanks for fixing.

**Comment by piskvorky** [comment]
@mpenkov can you check that the updated docs render OK? I still cannot access CircleCI, it's asking me for full access to my Github account + my soul.

**Review by piskvorky — APPROVED** [review]

**Comment by mpenkov** [comment]
Yes, the generated docs seem to be rendering just fine here: https://3588-1349775-gh.circle-artifacts.com/0/documentation/html/index.html

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -39,18 +39,22 @@
...
-    >>> path = get_tmpfile("word2vec.model")
-    >>>
     >>> model = Word2Vec(common_texts, size=100, window=5, min_count=1, workers=4)
     >>> model.save("word2vec.model")
 
-The training is streamed, meaning `sentences` can be a generator, reading input data
-from disk on-the-fly, without loading the entire corpus into RAM.
 
-It also means you can continue training the model later:
+The training is streamed, so `sentences` can be an iterable, reading input data
```
Comment (mpenkov) [inline gensim/models/word2vec.py:49]:
```suggestion
The training is streamed, so ``sentences`` can be an iterable, reading input data
```

**Comment by mpenkov** [comment]
Merged. Thank you for your contribution @polm !

**Pull request merged**